base/src/main/java/one/microstream/collections/lazy/LazyArrayList.java

```diff
@@ -0,0 +1,1367 @@
+ * By default, it will use the {@link LazySegmentUnloader.Default} that keeps two segments loaded.
+ *
+ * @param <E> the type of elements in this collection
+ */
+public class LazyArrayList<E> extends AbstractList<E> implements LazyList<E>, RandomAccess
+{
+	private static final int MAX_SEGMENT_SIZE_DEFAULT = 1000;
+	
+	///////////////////////////////////////////////////////////////////////////
+	// instance fields //
+	////////////////////
+	
+	private final int                 maxSegmentSize;
+	private final ArrayList<Segment>  segments      ;
+	private int                       size          ;
+	private final LazySegmentUnloader unloader      ;
+	
+	///////////////////////////////////////////////////////////////////////////
+	// constructors //
+	/////////////////
+	
+	
+	/**
+	 * Creates a new {@link LazyArrayList} with a maximum segment size of 1000.
+	 */
+	public LazyArrayList()
+	{
+		super();
+		this.maxSegmentSize = MAX_SEGMENT_SIZE_DEFAULT         ;
+		this.segments       = new ArrayList<>()                ;
+		this.unloader       = new LazySegmentUnloader.Default();
+	}
+	
+	
+	/**
+	 * Creates a new {@link LazyArrayList} with a defined maximum segment size.
+	 * 
+	 * @param maxSegmentSize maximum segment size, must be positive
+	 */
+	public LazyArrayList(final int maxSegmentSize)
+	{
+		super();
+		this.maxSegmentSize = positive(maxSegmentSize)         ;
+		this.segments       = new ArrayList<>()                ;
+		this.unloader       = new LazySegmentUnloader.Default();
+	}
+	
+	/**
+	 * Creates a new {@link LazyArrayList} with a defined maximum segment size.
+	 * 
+	 * @param maxSegmentSize maximum segment size, must be positive
+	 * @param lazySegmentUnloader LazySegmentUnloader instance
+	 */
+	public LazyArrayList(final int maxSegmentSize, final LazySegmentUnloader lazySegmentUnloader)
+	{
+		super();
+		this.maxSegmentSize = positive(maxSegmentSize);
+		this.segments       = new ArrayList<>()       ;
+		this.unloader       = lazySegmentUnloader     ;
+	}
+	
+	/**
+	 * Creates a new copy from the supplied {@link LazyArrayList}.
+	 * <br>
+	 * The Objects of the source list will not be copied,
+	 * both lists will reference the object instances.
+	 * 
+	 * @param list to be copied.
+	 */
+	public LazyArrayList(final LazyArrayList<E> list)
+	{
+		super();
+		this.maxSegmentSize = list.maxSegmentSize ;
+		this.segments       = new ArrayList<>()   ;
+		this.unloader       = list.unloader.copy();
+		this.addAll(list);
+	}
+	
+	///////////////////////////////////////////////////////////////////////////
+	// methods //
+	////////////
+	
+	/**
+	 * Returns the current number of internal segments.
+	 * 
+	 * @return the current number of internal segments.
+	 */
+    public long getSegmentCount()
+	{
+		return this.segments.size();
+	}
+    
+    /**
+     * Returns an Iterable over the Segment in this list.
+     * 
+     * @return an Iterable over the Segment in this list
+     */
+	public Iterable<? extends Segment> segments()
```

Comment:
format

Reply:
caused by spaces instead of tabs, corrected